### PR TITLE
#154953780 Show exercise info 

### PR DESCRIPTION
**Pull request by nyandiekaFelix**
#### What does this PR do?
 Add a read-only API endpoint that shows all infos for an exercise

#### Description of Task to be completed?
There should be a new read-only API endpoint that collects all the information of an exercise

Added endpoint:
```
GET /api/v2/exercise
GET /api/v2/exercise/<id>
```
#### How should this be manually tested?
- After setting up the project based on the instructions in the README,
run server with `python manage.py runserver`.
- Using postman , enter url `http://localhost:8000/api/v2/exerciseinfo/<id>`

#### What are the relevant pivotal tracker stories?
[154953780](https://www.pivotaltracker.com/story/show/154953780)

#### Screenshots 

screen shot 2018-02-19 at 11 47 05

### CODE REVIEW

File: wger/urls.py [hunk 1]
```diff
@@ -103,6 +103,7 @@
 
 # Exercises app
 router.register(r'exercise', exercises_api_views.ExerciseViewSet, base_name='exercise')
+router.register(r'exerciseinfo', exercises_api_views.ExerciseInfoViewSet, base_name='exerciseinfo')
```
Comment (muchai-mercy) [inline wger/urls.py]: Could you register the route as we mentioned, to take `exercise/<id>`

**Comment by muchai-mercy** [comment]
You may need to re-run `pycodestyle wger/` because of the tests failing? @nyandiekaFelix

**Comment by nyandiekaFelix** [comment]
@Mercy-Muchai  The Travis build is passing now. I fixed the coding style errors.

File: wger/exercises/api/views.py [hunk 2]
```diff
@@ -58,6 +59,45 @@ def perform_create(self, serializer):
...
+@api_view(['GET'])
+def exercise_info(request, id):
+    ''' Returns all information for an exercise '''
+
+    exercise_data = {}
+
+    exercise = get_object_or_404(Exercise, pk=id)
+
+    exercise_data['exercise'] = exercise
+    # exercise_data['languages'] = languages
```
Comment (muchai-mercy) [inline wger/exercises/api/views.py]: Do we need this?

File: wger/exercises/api/views.py [hunk 3]
```diff
@@ -58,6 +59,45 @@ def perform_create(self, serializer):
...
+    backgrounds_back.append('images/muscles/muscular_system_back.svg')
+
+    backgrounds = (backgrounds_front, backgrounds_back)
+    exercise_data['muscle_backgrounds_front'] = backgrounds[0]
+    exercise_data['muscle_backgrounds_back'] = backgrounds[1]
+
+    images = ['/media/' + str(image.image)
+              for image in exercise.exerciseimage_set.accepted()]
+    exercise_data['images'] = images
+    print(exercise_data)
```
Comment (muchai-mercy) [inline wger/exercises/api/views.py]: I don't think we need this print either

**Review by muchai-mercy — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 4]
```diff
@@ -58,10 +58,9 @@ target/
...
 .idea/
 
-# External Libraries 
+# External Libraries
 wger/core/static/bower_components
 node_modules
+
 .DS_Store
-venv-django
 .vscode/
```
Comment (muchai-mercy) [inline .gitignore]: Not sure you merged this file correct. Compare

File: wger/urls.py [hunk 5]
```diff
@@ -222,6 +222,9 @@
         nutrition_api_views.search,
         name='ingredient-search'),
     url(r'^api/v2/', include(router.urls)),
+    url(r'^api/v2/exercise/(?P<id>\d+)/$',
```
Comment (muchai-mercy) [inline wger/urls.py]: Any changes on this @nyandiekaFelix ?

Comment (nyandiekaFelix) [reply wger/urls.py]: I changed the parameter `name='exerciseinfo'` to `name='exercise'`. Anything else I need to change? @Mercy-Muchai

**Comment by muchai-mercy** [comment]
@nyandiekaFelix squash commits so it's ready for merge

**Comment by nyandiekaFelix** [comment]
@Mercy-Muchai  the commits have been squashed.

**Comment by muchai-mercy** [comment]
@nyandiekaFelix Fix failing test builds

**Comment by nyandiekaFelix** [comment]
@Mercy-Muchai  Build is passing now.

**Review by JoyyToo — APPROVED** [review]
LGTM

**Review by muchai-mercy — APPROVED** [review]
LGTM

**Pull request merged**